src/app/researcher/study/request/[memberSlug]/study-proposal.tsx

```diff
@@ -101,20 +101,56 @@ export const StudyProposal: React.FC<{ memberSlug: string }> = ({ memberSlug })
             })
             router.push(`/researcher/dashboard`)
         },
-        onError: async (error, _, context: { studyId: string; studyJobId: string } | undefined) => {
-            console.error(error)
+        onError: async (error: unknown, _variables, context: { studyId: string; studyJobId: string } | undefined) => {
+            console.error('Study proposal submission error:', error)
+
+            let title = 'Submission Failed' // Default title
+            let message = 'An unexpected error occurred.' // Default message
+            let needsCleanup = false
+            let rawErrorMessage = error instanceof Error ? error.message : 'Unknown error details'; // Capture raw message
+
+            // Check for specific access denied error
+            if (error instanceof Error && error.message.includes('You are not a member of this organization')) {
+                title = 'Access Denied'
+                message = 'You can only submit study proposals to organizations you are a member of.'
```

Comment:
isn't this case a non-issue? the `page.tsx` file already ensures the user is a member of the org 🤔